geoh5py/objects/geo_image.py

```diff
@@ -238,3 +252,134 @@ def vertices(self, xyz: np.ndarray | list):
         )
         self._vertices = xyz
         self.workspace.update_attribute(self, "vertices")
+
+    def georeferencing_from_tiff(self, image: Image.Image):
+        """
+        Get the geogrpahic information from the PIL image to georeference it.
+        Run the georefence() method of the object.
+        :param image: a .tif image open with PIL.Image.
+        """
+        try:
+            # get geographic information
+            georeferencing = {id_: image.tag[id_] for id_ in image.tag}
+            u_origin = georeferencing[33922][3]
+            v_origin = georeferencing[33922][4]
+            u_cell_size = georeferencing[33550][0]
+            v_cell_size = georeferencing[33550][1]
+            u_count = georeferencing[256][0]
+            v_count = georeferencing[257][0]
+            u_oposite = u_origin + u_cell_size * u_count
+            v_oposite = v_origin - v_cell_size * v_count
+
+            # prepare georeferencing
+            reference = np.array([[0.0, v_count], [u_count, v_count], [u_count, 0.0]])
+
+            locations = np.array(
+                [
+                    [u_origin, v_origin, 0.0],
+                    [u_oposite, v_origin, 0.0],
+                    [u_oposite, v_oposite, 0.0],
+                ]
+            )
+
+            # georeference the raster
+            self.georeference(reference, locations)
+
+        except (AttributeError, KeyError):
+            warn("The 'tif.' image has no referencing information.")
+
+    def to_grid2d(
+        self,
+        name: str = None,
+        rotation: float = 0.0,
+        dip: float = 0.0,
+        elevation: float = 0.0,
+        transform: str = "GRAY",
+        parent: Group = None,
```

Comment:
As discussed, let's pass `**grid2d_kwargs` from here to the `.create(**grid2d_kwargs)` instead and remove `name`, `rotation`, `dip` and `parent` from this list.